restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/CallsEndpoint.java

```diff
@@ -228,7 +236,82 @@ protected Response getCalls(final String accountSid, UriInfo info,
         String endTime = info.getQueryParameters().getFirst("EndTime");
         String parentCallSid = info.getQueryParameters().getFirst("ParentCallSid");
         String conferenceSid = info.getQueryParameters().getFirst("ConferenceSid");
-        String reverse = info.getQueryParameters().getFirst("Reverse");
+
+        // 'Reverse' URL parameter currently behaves a bit weird; it has no control over the direction of the ordering,
+        // as is the case for example in IncomingPhoneNumbersEndpoint. What it does is affecting paging instead,
+        // and the reason I think is that back when this was implemented to avoid too much work on BE we wanted Console
+        // to be able to get the right set of results, even if without ordering so that they are ordered in
+        // the client side.
+
+        // But now that sorting is implemented, I think we don't need that anymore and we should be ok ignoring it. Also,
+        // I believe having a boolean Reverse url parameter won't be able to scale well in the future if we introduce
+        // multiple SortBy fields for example.
+        //String reverse = info.getQueryParameters().getFirst("Reverse");
+
+        // Format for sorting URL paremeter is '?SortBy=<field>:<direction>', for example: '?SortBy=start_time:asc'. In the
+        // future we can extend it to multiple sort fields, like ?SortBy=<field1>:<direction1>,<field2>:<direction2>
+        String sortParameters = info.getQueryParameters().getFirst("SortBy");
+
+        CallDetailRecordFilter.Builder filterBuilder = CallDetailRecordFilter.Builder.builder();
+
+        String sortBy = null;
+        String sortDirection = null;
+
+        if (sortParameters != null && !sortParameters.isEmpty()) {
+            final String[] values = sortParameters.split(":", 2);
+            sortBy = values[0];
+            if (values.length > 1) {
+                sortDirection = values[1];
+                if (sortBy.isEmpty()) {
+                    return status(BAD_REQUEST).entity(buildErrorResponseBody("Error parsing the SortBy parameter: missing field to sort by", responseType)).build();
+                }
+                if (!sortDirection.equalsIgnoreCase("asc") && !sortDirection.equalsIgnoreCase("desc")) {
+                    return status(BAD_REQUEST).entity(buildErrorResponseBody("Error parsing the SortBy parameter: sort direction needs to be either \'asc\' or \'desc\'", responseType)).build();
+                }
+            }
+        }
+
+        ///////
+        if (sortBy != null) {
+            if (sortBy.equals("date_created")) {
+                if (sortDirection != null) {
+                    if (sortDirection.equalsIgnoreCase("asc")) {
+                        filterBuilder.sortedByDate(SortDirection.ASCENDING);
+                    } else {
+                        filterBuilder.sortedByDate(SortDirection.DESCENDING);
+                    }
+                }
+            }
+
+            if (sortBy.equals("from")) {
```

Comment:
is "from" or "From"?

Reply:
@jaimecasero yes, please check my above answer on this.

Also, another thing that would be nice is to have the attribute names (like DateCreated) stored somewhere as constants so that we can reuse in multiple places (like above, in the converter, etc) and make sure we don't mess up the names. We could declare them for example in CallDetailRecord.java for example? Seems like the object model somehow. Wdyt?